std/protocols/permutation.asm

```diff
@@ -122,23 +101,16 @@ let permutation: expr[], Constr -> Constr[] = |acc, permutation_constraint| {
     let (lhs_selector, lhs, rhs_selector, rhs) = unpack_permutation_constraint(permutation_constraint);
 
     let _ = assert(len(lhs) == len(rhs), || "LHS and RHS should have equal length");
-    let with_extension = match len(acc) {
-        1 => false,
-        2 => true,
-        _ => panic("Expected 1 or 2 accumulator columns!")
-    };
 
-    let _ = if !with_extension {
-        assert(!needs_extension(), || "The Goldilocks field is too small and needs to move to the extension field. Pass two accumulators instead!")
-    } else { () };
+    let with_extension = is_extension(len(acc));
+    let _ = assert_extension(with_extension);
 
     // On the extension field, we'll need two field elements to represent the challenge.
     // If we don't need an extension field, we can simply set the second component to 0,
     // in which case the operations below effectively only operate on the first component.
-    let fp2_from_array = |arr| if with_extension { Fp2::Fp2(arr[0], arr[1]) } else { from_base(arr[0]) };
-    let acc_ext = fp2_from_array(acc);
-    let alpha = fp2_from_array([alpha1, alpha2]);
-    let beta = fp2_from_array([beta1, beta2]);
+    let acc_ext = fp2_from_array(acc, with_extension);
```

Comment:
This function call should not truncate acc, because you determine if you work on the extension field depending on the length of acc. Shouldn't we rather pass two accumulator columns to this function and then ignore the second one depending on the well-known field?

Reply:
Yes, `with_extension` parameter looks redundant